guides/source/caching_with_rails.md

```diff
@@ -370,6 +370,18 @@ There are some common options that can be used by all cache implementations. The
 
 * `:race_condition_ttl` - This option is used in conjunction with the `:expires_in` option. It will prevent race conditions when cache entries expire by preventing multiple processes from simultaneously regenerating the same entry (also known as the dog pile effect). This option sets the number of seconds that an expired entry can be reused while a new value is being regenerated. It's a good practice to set this value if you use the `:expires_in` option.
 
+When using Puma or another threaded server it is recommended to use a connection pool to prevent thread contention. Connection pooling is only supported by the `MemCacheStore` and `RedisCacheStore`. To enable the connection pool first add the `connection_pool` gem to your Gemfile.
```

Comment:
> it is recommended to use a connection pool to prevent thread contention

This won't mean anything to the reader unless they already understand what a connection pool does. Googling "thread contention" is likely to scare off anyone without a computer science background.

I don't think we need to go quite this far, but the Configuring Rails Applications guide has a more detailed explanation of why Active Record's connection pool exists and how it works:

https://github.com/rails/rails/blob/8eef8791fab0985060e7a17687698383b3d47c2c/guides/source/configuring.md#database-pooling

Reply:
I've moved the `connection pool options` to a separate subsection, made the suggested changes, and changed the `thread contention` recommendation to a suggestion:
```
By default the `MemCacheStore` and `RedisCacheStore` use a single connection
per process. This means that if you're using Puma, or another threaded server,
you can have multiple threads waiting for the connection to become available.
To increase the amount of available connections you can enable connection
pooling.
```